cpp/src/arrow/dataset/file_parquet.h

```diff
@@ -91,6 +91,10 @@ class ARROW_DS_EXPORT ParquetFileFormat : public FileFormat {
     /// @{
     std::unordered_set<std::string> dict_columns;
     /// @}
+
+    /// Parallelize conversion across columns. This option is ignored if a scan is already
+    /// parallelized across input files.
```

Comment:
Add note that this is experimental. Once we fix our nested parallelism problem this shouldn't be needed